### PR TITLE
This PR, Adds the following:

### PR DESCRIPTION
**Pull request by Khaledgarbaya**
* Copy and EditorInterface from one field to another
* Reset an EditorInterface of a field

* When changing a field id, the EditorInterface that is associated to it is now copied to the new ID

TODO:

- [x] Update README
- [x] Update typing

### CODE REVIEW

File: src/lib/intent/editorinterface-reset.ts [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+    return true
+  }
+  shouldSave (): boolean {
+    return false
+  }
+  shouldPublish (): boolean {
+    return false
+  }
+  toActions () {
+    // TODO: check for optimizing API calls and saving multiple editor interfaces changes on the same content type done as 1 single API call instead of multiple.
```
Comment (TimBeyer) [inline src/lib/intent/editorinterface-reset.ts]: That's what `groupsWith` does.
Everything that `groupsWith` returns true for will be turned into a single request later on.
You will ave to set `endsGroup` to `false` though

File: test/unit/lib/intent/editorinterface-update.spec.ts [hunk 2]
```diff
@@ -28,22 +28,28 @@ describe('EditorInterfaceUpdateIntent', function () {
         ct.changeEditorInterface('name', 'markdown')
           .changeEditorInterface('description', 'singleLine')
       })
-
+      debugger
```
Comment (TimBeyer) [inline test/unit/lib/intent/editorinterface-update.spec.ts]: Probably should be removed

File: src/lib/intent/editorinterface-reset.ts [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+import chalk from 'chalk'
+import { SaveEditorInterfaceAction } from '../action/editorinterface-save'
+
+export default class EditorInterfaceResetIntent extends Intent {
+  isEditorInterfaceReset () {
+    return true
+  }
+  groupsWith (other: Intent): boolean {
+    const sameContentType = other.getContentTypeId() === this.getContentTypeId()
+    return other.isEditorInterfaceReset() && sameContentType
```
Comment (TimBeyer) [inline src/lib/intent/editorinterface-reset.ts]: Maybe these also group with `editorInterfaceCopy` and the other EI intents?

File: src/lib/intent/editorinterface-copy.ts [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+export default class EditorInterfaceCopyIntent extends Intent {
+  isEditorInterfaceCopy () {
+    return true
+  }
+  groupsWith (other: Intent): boolean {
+    const sameContentType = other.getContentTypeId() === this.getContentTypeId()
+    return other.isEditorInterfaceCopy() && sameContentType
+  }
+  endsGroup (): boolean {
+    return true
```
Comment (TimBeyer) [inline src/lib/intent/editorinterface-copy.ts]: I guess copys are also groupable?

File: test/unit/lib/offline-api/request-batches.spec.ts [hunk 5]
```diff
@@ -227,7 +227,7 @@ describe('Payload builder', function () {
       ]
 
       const requests = await buildPayloads(migration, existingCts)
-
+      debugger
```
Comment (TimBeyer) [inline test/unit/lib/offline-api/request-batches.spec.ts]: Another debugger

File: src/lib/intent/composed-intent.ts [hunk 6]
```diff
@@ -4,6 +4,7 @@ import { PlanMessage, Section } from '../interfaces/plan-message'
 import { difference, groupBy, flatten, values, entries } from 'lodash'
 
 function mergeSections (sections: Section[]): Section {
+  debugger
```
Comment (phoebeschmidt) [inline src/lib/intent/composed-intent.ts]: whoops :D

**Pull request merged**